src/main.rs

```diff
@@ -144,6 +165,55 @@ async fn main() -> Result<()> {
 		TriggerRepository,
 	>(None, None, None)?;
 
+	let monitor_path = matches
```

Comment:
Let's separate all this logic out from the `main` function for better readability (we can still keep it in the same file):
Something like: `should_test_monitor_execution` and `test_monitor_execution` and then call it here.

```
if should_test_monitor_execution(..) {
    test_monitor_execution(...) -> Result()
}
```

Reply:
Let's also add some comments to state these are 2 different execution paths (one is to test a monitor, the other is to run the service)